src/bundle/ui-dev/src/modules/common/popup/popup.component.js

```diff
@@ -16,205 +18,128 @@ const MODAL_SIZE_CLASS = {
+        if (isVisible) {
+            showPopup();
+            modalRef.current.addEventListener('hidden.bs.modal', onClose);
         }
-
-        return (
-            <div {...modalAttrs}>
-                <div className={`modal-dialog c-popup__dialog ${MODAL_SIZE_CLASS[size]}`} role="dialog">
-                    <div className="modal-content c-popup__content">
-                        {noHeader ? this.renderCloseButton() : this.renderHeader()}
-                        <div className="modal-body c-popup__body">{this.props.children}</div>
-                        {this.renderFooter()}
+    }, [isVisible]);
+
+    return (
+        <div ref={modalRef} className={modalClasses} tabIndex={hasFocus ? -1 : undefined}>
+            <div className={`modal-dialog c-popup__dialog ${MODAL_SIZE_CLASS[size]}`} role="dialog">
+                <div className="modal-content c-popup__content">
+                    {noHeader
+                        ? closeButton
+                        : title && (
+                              <div className={'modal-header c-popup__header'}>
+                                  <h3 className="modal-title c-popup__headline" title={title}>
+                                      <span className="c-popup__title">{title}</span>
+                                      {subtitle && <span className="c-popup__subtitle">{subtitle}</span>}
+                                  </h3>
+                                  {closeButton}
+                              </div>
+                          )}
+                    <div className="modal-body c-popup__body">{children}</div>
+                    <div className={'modal-footer c-popup__footer'}>
+                        {actionBtns.map(({ className, onClick, disabled = false, label, ...extraProps }) => (
+                            <button
+                                key={label}
+                                type="button"
+                                className={`btn ibexa-btn ${className}`}
+                                onClick={onClick ? (event) => handleOnClick(event, onClick) : hidePopup}
+                                disabled={disabled}
+                                {...extraProps}
+                            >
+                                {label}
+                            </button>
+                        ))}
                     </div>
                 </div>
             </div>
-        );
-    }
-}
+        </div>
+    );
+};
 
 Popup.propTypes = {
-    isVisible: PropTypes.bool,
-    isLoading: PropTypes.bool,
-    onClose: PropTypes.func.isRequired,
-    children: PropTypes.element.isRequired,
+    actionBtns: PropTypes.arrayOf(
+        PropTypes.shape({
+            label: PropTypes.string.isRequired,
+            onClick: PropTypes.func,
+            disabled: PropTypes.bool,
+            className: PropTypes.string,
+        }),
+    ).isRequired,
+    children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]).isRequired,
```

Comment:
```suggestion
    children: PropTypes.node.isRequired,
```
is enough as far as I remember